liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -196,6 +199,40 @@ unoptimiseDynFlags df = updOptLevel 0 df
     , ghcMode      = CompManager
     }
 
+--------------------------------------------------------------------------------
+-- | Parsing phase -------------------------------------------------------------
+--------------------------------------------------------------------------------
+
+-- | We hook at this stage of the pipeline to store the parsed module
+-- for later processing during typechecking:
+--
+-- * Comments are preserved during parsing because we turn on
+--   'Opt_KeepRawTokenStream'. So we can get the /LH/ specs from there.
```

Comment:
Note that different flags are set at two different times. There is `lhDynFlags`, set before the plugin. And there is `customDynFlags` set only for parsing, typechecking, and desugaring.

The benefit of `customDynFlags` is that it doesn't affect anything else in the compiler for LH sake. But I'm guessing that you want `Opt_KeepRawTokenStream` to affect how GHC parses all files, and then you will have to move it to `lhDynFlags`.